### ISSUE TITLE
incorrect usage of cosine annealing

### ISSUE THREAD

**Issue by qihuifengrwth** [AmirmohammadRostami/KeywordsSpotting-EfficientNet-A0#1]
Hi,

We are currently using your code.Your model also tackle with Chinese KWS pretty well and we appreciate your work. However, we found out there could be a mistake in your scheduling tactics which may have an effect on the final performance.

You applied cosine annealing schedule, then the lr is supposed to go down and up for several iterations. However, in your default settings, the initial learning rate and the minimum learning (eta_min) are both 1e-3. As a result, the actually lr is 1e-3 during the entire training process. Besides, the parameter T_max should refer to half of the cosine period. As you set T_max to be the number of epochs, the actual lr would be "cosine decaying" but not periodic. I'm not sure whether that is exactly the effect you want...

kind regards,
Nelson

**Comment by AmirmohammadRostami** [comment]
Hi Nelsonvon,
We glad that model obtained good results in Chinese KWS as well as Persian, if you have published your results somewhere, please let us know.
You are right about the cosine annealing scheduler and its parameters. Our goal was to reduce the learning rate with something like the cosine function. So, we simply implemented it with the cosine annealing scheduler and it's worked well for our purpose.
